File: AndroidThermostat/ExositeThermostat/src/main/java/com/exosite/app/MainActivity.java
package com.exosite.app;

import android.support.v4.app.FragmentTransaction;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import com.crashlytics.android.Crashlytics;

import cz.saymon.exosite.weatherstation.R;
import io.fabric.sdk.android.Fabric;

public class MainActivity extends AppCompatActivity {
    private static final String TAG = MainActivity.class.getSimpleName();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Fabric.with(this, new Crashlytics());
        setContentView(R.layout.activity_main);

        if (savedInstanceState == null) {
            PlaceholderFragment frag = new PlaceholderFragment();
            FragmentTransaction trans = getSupportFragmentManager().beginTransaction();
            trans.add(R.id.container, frag, PlaceholderFragment.FRAGMENT_TAG).commit();
        }
    }

}

File: ExositeOnePlatformRpcTest/app/src/main/java/cz/saymon/android/exositeoneplatformrpctest/ui/activities/MainActivity.kt
package cz.saymon.android.exositeoneplatformrpctest.ui.activities

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import cz.saymon.android.exositeoneplatformrpctest.R
import cz.saymon.android.exositeoneplatformrpctest.model.data_objects.Bar
import cz.saymon.android.exositeoneplatformrpctest.model.data_objects.Foo
import cz.saymon.android.exositeoneplatformrpctest.model.retrofit.ServerApi
import cz.saymon.android.exositeoneplatformrpctest.model.retrofit.ServerRequest.ServerRequest
import cz.saymon.android.exositeoneplatformrpctest.model.retrofit.ServerResponse.ServerResponse
import cz.saymon.android.exositeoneplatformrpctest.utils.app
import cz.saymon.android.exositeoneplatformrpctest.utils.toast
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.disposables.Disposable
import kotlinx.android.synthetic.main.activity_main.*
import timber.log.Timber
import javax.inject.Inject

// If we want to call the synthetic properties on View (useful in adapter classes)
//import kotlinx.android.synthetic.main.activity_main.view.*

class MainActivity : AppCompatActivity() {

    @Inject
    lateinit var api: ServerApi

    internal var subscription: Disposable? = null
    internal var serverRequest = ServerRequest()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        app.appComponent.inject(this)

        main.setOnClickListener(this::testApi);

        Timber.d("onCreate()")

        toast("Initialized")

        val test = Foo("foo test", Bar(42))
        Timber.d(test.toString())
    }

    private fun testApi(_ignored: View) {
        subscription?.dispose()

        subscription = api.getItem(serverRequest)
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(
                        { dataset -> handleResponse(dataset) },
                        { throwable -> handleResponse(throwable) })
    }

    private fun handleResponse(dataset: List<ServerResponse>?) {
        toast("onNext: ${System.currentTimeMillis()}")
        Timber.d(dataset.toString())
        text.text = dataset.toString()
    }

    private fun handleResponse(throwable: Throwable?) {
        toast("onException: ${System.currentTimeMillis()}")
        text.text = throwable.toString()
    }

}
